tensorflow/core/kernels/cwise_ops_common.h

```diff
@@ -516,6 +517,15 @@ struct ApproximateEqual<CPUDevice, T> {
 // Instead of adding REGISTER10, etc., shard the .cc files - see
 // cwise_op_equal_to_*.cc for an example.
 
+// add half float when necessary
+#ifdef TENSORFLOW_SYCL_NO_DOUBLE
```

Comment:
In that case it feels like better place for that will be: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/register_types.h#L183